lib/iris/fileformats/pp.py

```diff
@@ -1644,10 +1644,6 @@ def __eq__(self, other):
                 if all(attrs):
                     self_attr = getattr(self, attr)
                     other_attr = getattr(other, attr)
-                    if isinstance(self_attr, biggus.NumpyArrayAdapter):
```

Comment:
Why do we think we no longer need a realisation step here 
 -- as in "if is_lazy_data(x): x=as_concrete_data(x)" for both attrs ??